### DISCUSSION TITLE
config that user can only create a doc and not update or delete it

### DISCUSSION THREAD

**Discussion by sergiupopescu199** [apache/couchdb#3512] [Q&A]
Hi, I created  a non admin user, I struggle on how to config the database to allow only the creation of docs by users but not the update or deletion
Could someone help me, thanks in advance

**Comment by jjrodrig** [comment]
Hi @sergiupopescu199,

I think that the mechanism for achieving this functionality is by defining a Validate Document Update Function per database.  
You can check the doc here - https://docs.couchdb.org/en/stable/ddocs/ddocs.html#validate-document-update-functions

I hope this helps
